crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -297,6 +298,31 @@ impl<T: TransportConnect> Scheduler<T> {
                 let mut modified = false;
 
                 match replication_strategy {
+                    ReplicationStrategy::OnOneNodeNoFollowers => {
+                        // we always take the first worker node
+                        // regardless its alive or not
+                        // this is because local/in-memory loglets
+                        // can't be migrated and has to colocate on the
+                        // same node
+                        let first_worker = nodes_config
+                            .iter()
+                            .filter_map(|(node_id, node)| {
+                                if node.roles.contains(Role::Worker) {
+                                    Some(node_id)
+                                } else {
+                                    None
+                                }
+                            })
+                            .sorted()
+                            .next()
+                            .into_iter()
+                            .collect();
```

Comment:
Technically speaking, there is no guarantee that we won't register first a node with node_id `1` and then a node with node_id `0` (e.g. when forcing the node_ids). Hence, we might still move partition processors around whenever a node with a lower node_id registers. Since this solution is not bullet proof and we probably will soon have the replicated loglet as the default, logging a warning when using the local loglet with multiple nodes as Ahmed suggested might be the simplest temporary bandaid until we have the replicated loglet as the default.

Reply:
Okay, sounds good to me